src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java

```diff
@@ -0,0 +1,116 @@
+package org.mozilla.magnet.net.scanner.geolocation;
+
+import android.content.Context;
+import android.location.Location;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.GoogleApiClient.OnConnectionFailedListener;
+import com.google.android.gms.common.api.GoogleApiClient.ConnectionCallbacks;
+import com.google.android.gms.location.LocationServices;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.mozilla.magnet.net.scanner.BaseScanner;
+import org.mozilla.magnet.net.scanner.MagnetScannerItem;
+
+public class ScannerGeolocation extends BaseScanner implements ConnectionCallbacks, OnConnectionFailedListener {
+    private final static String TAG = "ScannerGeolocation";
+    private final static String SCANNER_TYPE = "geolocation";
+    private final static String SLUG_BASE_URL = "https://tengam.org/";
+    private final static String API_SEARCH_URL = "https://tengam.org/content/v1/search/beacons/";
+    private final static int SCAN_RADIUS = 10;
+    private GoogleApiClient mGoogleApiClient;
+    private RequestQueue mQueue;
+
+    public ScannerGeolocation(Context context) {
+        mQueue = Volley.newRequestQueue(context);
+
+        mGoogleApiClient = new GoogleApiClient.Builder(context)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .addApi(LocationServices.API)
+                .build();
+    }
+
+
+    @Override
+    public String scannerType() {
+        return SCANNER_TYPE;
+    }
+
+    @Override
+    protected void doStart() {
+        mGoogleApiClient.connect();
+    }
+
+    @Override
+    public void stop() {
+        mGoogleApiClient.disconnect();
+    }
+
+    @Override
+    public void onConnected(Bundle connectionHint) {
+        Location location = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
+        Log.d(TAG, "lat: " + location.getLatitude());
+        Log.d(TAG, "lon: " + location.getLongitude());
+        scan(location);
```

Comment:
Should we check the location accuracy before performing a scan?

location.getAccuracy()

https://developer.android.com/reference/android/location/Location.html#getAccuracy()